tests/logging_tests.cpp

```diff
@@ -0,0 +1,72 @@
+#include <boost/test/unit_test.hpp>
+#include <boost/chrono.hpp>
+#include <boost/thread/thread.hpp>
+
+#include <fc/thread/thread.hpp>
+#include <fc/reflect/reflect.hpp>
+#include <fc/reflect/variant.hpp>
+#include <fc/log/file_appender.hpp>
+#include <fc/log/logger.hpp>
+#include <fc/log/logger_config.hpp>
+#include <fc/variant.hpp>
+#include <fc/time.hpp>
+#include <fc/io/json.hpp>
+#include <fc/io/fstream.hpp>
+
+#include <thread>
+#include <iostream>
+#include <fstream>
+
+BOOST_AUTO_TEST_SUITE(logging_tests)
+
+BOOST_AUTO_TEST_CASE(log_reboot)
+{
+    BOOST_TEST_MESSAGE("Setting up logger");
+    fc::file_appender::config conf;
+    conf.filename = "/tmp/my.log";
+    conf.format = "${timestamp} ${thread_name} ${context} ${file}:${line} ${method} ${level}]  ${message}";
+    conf.flush = true;
+    conf.rotate = true;
+    conf.rotation_interval = fc::seconds(5); // rotate every 5 seconds
+    conf.rotation_limit = fc::seconds(20); // Don't keep files older than 20 seconds
+    conf.max_object_depth = 200;
+
+    fc::appender::ptr fa = fc::appender::create("file", "file", fc::variant(conf, 200));
+
+    fc::path prev_log_filename = "";
+
+    BOOST_TEST_MESSAGE("Starting Loop");
+    for(int i = 0; i < conf.rotation_limit.to_seconds(); i++)
+    {
+        fc::log_context ctx(fc::log_level::all, "my_file.cpp", i, "my_method()");
+        fc::log_message my_log_message( ctx, "${message}", {"message","This is a test"} );
+        fa->log(my_log_message);
+
+        fc::time_point now = fc::time_point::now();
+        int64_t interval_seconds = conf.rotation_interval.to_seconds();
+        int64_t file_number = now.sec_since_epoch() / interval_seconds;
+        fc::time_point_sec start_time = fc::time_point_sec( (uint32_t)(file_number * interval_seconds) );
+        fc::string timestamp_string = start_time.to_non_delimited_iso_string();
+        fc::path link_filename = conf.filename;
+        fc::path log_filename = link_filename.parent_path() / (link_filename.filename().string() + "." + timestamp_string);
```

Comment:
I believe this could be simplified if we added a member to the file_appender that is the current filename.

@abitmore what do you think? Is such a change advisable? Is it worth it? Would it be considered "scope creep"?

Advantage: Easier testing, elimination of duplicate code.
Disadvantage: It could encourage someone to open the file and add their own information in the log.
Disadvantage: This test would have to cast the fc::appender::ptr to a file_appender to get the value.
Disadvantage: By the time the test gets the filename, it could be stale due to the multithreaded nature of this appender. Although the code above suffers from the same problem.

Reply:
Sorry, I don't have an idea.

Reply:
should I do this change @jmjatlanta ? guys what do you think ?

Reply:
Disadvantage: By the time the test gets the filename, it could be stale due to the multithreaded nature of this appender. Although the code above suffers from the same problem. - Why current test code has the same problem ? Doesn't this test code run in it's own thread to check how appender works ?

Reply:
so should I change this snippet of code or not ?

Reply:
@jmjatlanta should I change that code or not ? is this issue finished or not yet ?

Reply:
I don't think changing the code is worth it, as the problem will exist regardless. There is a very slim chance that the test will fail, but I doubt anyone will run into it. I will review everything again. If the chance of failure is there, perhaps we should document it as a comment within the test.